### PR TITLE
Adds support for the computer field in Windows Eventlogs

### PR DESCRIPTION
**Pull request by defensivedepth**
Parses out the `Computer` field for Windows Eventlogs, which is important as it is the hostname of the system on which the event was originally generated.

Partially completes https://github.com/osquery/osquery/issues/6726

### CODE REVIEW

**Comment by mike-myers-tob** [comment]
To pass the code style enforcement check, you can run `clang_format -i /path/to/source_code.cpp` on the files you have changed previously, and then commit/push those changes.

**Review by mike-myers-tob — COMMENTED** [review]
The changes look good to me.

I built and ran on Windows 10 and it works well with some quick testing.

`osqueryi.exe --disable_events=false --windows_event_channels="System,Application,Setup,Security" --enable_windows_events_publisher=true --enable_windows_events_subscriber=true`

`osquery> select * from windows_events LIMIT 10;`

**Comment by mike-myers-tob** [comment]
It looks like the newly added test case is failing on the CI because certain mock event test data ([JSON files, here](https://github.com/osquery/osquery/tree/master/osquery/tables/events/tests/windows/data/windows_events/output/Application)) do not have the `computername` field.

To re-run just the failing test locally:

`ctest -R osquery_tables_events_tests_windowseventstests-test -C RelWithDebInfo -V`

**Comment by directionless** [comment]
> It looks like the newly added test case is failing on the CI because certain mock event test data ("recorded events") (JSON files, here) do not have the computername field.

Is this an error in the test data, or is this field sometimes missing?

**Comment by mike-myers-tob** [comment]
> > It looks like the newly added test case is failing on the CI because certain mock event test data ("recorded events") (JSON files, here) do not have the computername field.
> 
> Is this an error in the test data, or is this field sometimes missing?

@alessandrogario you originally put these test files together, right? Were they manually exported from the event log selecting only certain fields?

**Comment by mike-myers-tob** [comment]
> @alessandrogario you originally put these test files together, right? Were they manually exported from the event log selecting only certain fields?

In a separate conversation with me he said that he exported these Windows events manually from Windows Event Viewer or possibly `wevtutil` as XML. Then he manually edited them, removing fields to create these test files, which he then ran through osquery to create the JSON files and manually reviewed them for correctness.

Hopefully that answers the question of why this particular field is missing.

**Comment by defensivedepth** [comment]
Should I manually add that field to the test files?

**Comment by mike-myers-tob** [comment]
> Should I manually add that field to the test files?

Yea, that's all that is needed. I manually confirmed that with a couple of them. It will unfortunately be a little tedious to update them all manually, but you can edit each XML file with an `xmlstarlet` one-liner like this:

```
cd ./osquery/tables/events/tests/windows/data/windows_events
xmlstarlet edit -L --subnode "/Event/System" -t elem -n "Computer" -v "DESKTOP-4AR7BIA" ./input/Application/1.xml
```

and then each JSON file with `jq` like this:

`jq '. += {"computer_name": "DESKTOP-4AR7BIA"}' ./output/Application/1.json`

Except `jq` can't edit the file in-place so you have to append that with, like, `> JSON.tmp && mv JSON.tmp ./output/Application/1.json`

**Comment by defensivedepth** [comment]
not a problem, will get it done and submitted. Thanks for the guidance!

**Comment by mike-myers-tob** [comment]
Looks like that JSON parsing test is having a problem still

https://github.com/osquery/osquery/pull/6952/checks?check_run_id=1989915214#step:17:2239

**Comment by defensivedepth** [comment]
My mistake, I pushed to the wrong branch. Still cleaning up the test files.

**Comment by defensivedepth** [comment]
Local tests are now passing. I had to edit a few of the source test files as they had different hostnames - normalized them all to `DESKTOP-4AR7BIA`

**Comment by defensivedepth** [comment]
For those who may need to bulk edit these test source files in the future (Will need to tweak for your use case):

XML:  `xmlstarlet edit -L --subnode "/Event/System" -t elem -n "Computer" -v "DESKTOP-4AR7BIA" ./input/*/*.xml`

JSON: `for f in output/*/*.json;  do jq '. += {"computer_name": "DESKTOP-4AR7BIA"}' ${f} | sponge ${f}; done;`

**Review by mike-myers-tob — APPROVED** [review]

**Review by theopolis — APPROVED** [review]

File: specs/windows/windows_eventlog.table [hunk 1]
```diff
@@ -7,6 +7,7 @@ schema([
     Column("level", INTEGER, "Severity level associated with the event"),
     Column("provider_name", TEXT, "Provider name of the event"),
     Column("provider_guid", TEXT, "Provider guid of the event"),
+    Column("computer_name", TEXT, "Hostname of system where event was generated"),
```
Comment (VikasBhumca2006) [inline specs/windows/windows_eventlog.table:10]: Updated as part of PR "[Adds support for the computer field in Windows Eventlogs
#6952](https://github.com/osquery/osquery/pull/6952/files#diff-dd760b19aa7115cbd9a4bbe4b85efffdff2097c683483a2090bb59b3a3e42229R10)

File: osquery/tables/system/windows/windows_eventlog.cpp [hunk 2]
```diff
@@ -53,6 +53,7 @@ Status parseWelXml(QueryContext& context, std::wstring& xml_event, Row& row) {
   row["channel"] = SQL_TEXT(windows_event.source);
   row["provider_name"] = SQL_TEXT(windows_event.provider_name);
   row["provider_guid"] = SQL_TEXT(windows_event.provider_guid);
+  row["computer_name"] = SQL_TEXT(windows_event.computer_name);
```
Comment (VikasBhumca2006) [inline osquery/tables/system/windows/windows_eventlog.cpp:56]: Added as part of PR : https://github.com/osquery/osquery/pull/6952/files#

**Pull request merged**